### ISSUE TITLE
Source File throws exception "Duplicate names are not allowed" 

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#2331]
## Current Behavior
A user has reported that the source file breaks when they feed it the attached file below: 
[aws_s3_sample.csv.zip](https://github.com/airbytehq/airbyte/files/6092001/aws_s3_sample.csv.zip)

They get the following error: 
```
airbyte-scheduler | 2021-03-05 04:21:43 INFO i.a.w.p.a.DefaultAirbyteStreamFactory(internalLog):110 - {job_id=2425, job_log_filename=logs.log, job_root=/tmp/workspace/2425/0} - Discovering schema of aws_s3_test at s3://swayam-nc-connector-test-v1/aws_s3_sample.csv...
airbyte-scheduler | 2021-03-05 04:21:43 ERROR i.a.w.p.a.DefaultAirbyteStreamFactory(internalLog):108 - {job_id=2425, job_log_filename=logs.log, job_root=/tmp/workspace/2425/0} - Failed to discover schemas of aws_s3_test at s3://swayam-nc-connector-test-v1/aws_s3_sample.csv: ValueError('Duplicate names are not allowed.')
airbyte-scheduler | Traceback (most recent call last):
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/source_file/source.py", line 121, in discover
airbyte-scheduler |     streams = list(client.streams)
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/source_file/client.py", line 339, in streams
airbyte-scheduler |     "properties": self._stream_properties(),
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/source_file/client.py", line 325, in _stream_properties
airbyte-scheduler |     df_list = self.load_dataframes(fp, skip_data=False)
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/source_file/client.py", line 283, in load_dataframes
airbyte-scheduler |     return [reader(fp, **reader_options)]
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/pandas/io/parsers.py", line 605, in read_csv
airbyte-scheduler |     return _read(filepath_or_buffer, kwds)
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/pandas/io/parsers.py", line 454, in _read
airbyte-scheduler |     _validate_names(kwds.get("names", None))
airbyte-scheduler |   File "/usr/local/lib/python3.7/site-packages/pandas/io/parsers.py", line 435, in _validate_names
airbyte-scheduler |     raise ValueError("Duplicate names are not allowed.")
airbyte-scheduler | ValueError: Duplicate names are not allowed.
airbyte-scheduler |
airbyte-scheduler | 2021-03-05 04:21:43 ERROR i.a.c.i.LineGobbler(voidCall):69 - {job_id=2425, job_log_filename=logs.log, job_root=/tmp/workspace/2425/0} - Traceback (most recent call last):
```

## Steps to Reproduce
1. Setup a file source connection with the provided file as an input CSV (make sure to unzip it first)
1. Setup any destination
1. Run a sync
2. 💣 

## Severity of the bug for you
High - blocking usage of the file source connector 

## Airbyte Version
0.14.3-alpha

## Connector Version (if applicable)
unknown

## Additional context
Slack thread: https://airbytehq.slack.com/archives/C01MFR03D5W/p1614918257159100

**Comment by yevhenii-ldv** [comment]
@sherifnada 
I tried to reproduce this error with a csv file, but I could not reproduce it.
Initially, I tried to do it locally without using UI and I successfully ran the `discover `and `read `commands for the `aws_s3_sample.csv` file.
Then I used YUI and with it I already downloaded the file, this time I used Amazon Web Service and the following link:
`s3://swayam-nc-connector-test-v1/aws_s3_sample.csv`

I asked @vitaliizazmic to try to read the given csv file and he confirmed that the reading was going without problems.

Investigating the forums, I came across the fact that there really is such a problem with pandas in later versions, and it is recommended to roll back the library version to 0.20.0 (we are currently using 1.2.0). This can be done, but at the moment I have not received any problems in order to read the proposed csv file.

**Comment by yevhenii-ldv** [comment]
I tried to run the command:
`docker run --rm -v "$(pwd)/airbyte-integrations/connectors/source-file/secrets:/secrets" airbyte/source-file:0.1.10 read --config /secrets/config.json --catalog catalog.json`

This gave the same result, did not find any errors

**Comment by sherifnada** [comment]
@yevhenii-ldv thanks for the thorough checks. Let's wait for the user to get back to us regarding the connector version.

**Comment by sherifnada** [comment]
- @yevhenii-ldv please see the slack message from the user. Looks like the difference is the reader options. 

https://airbytehq.slack.com/archives/C01MFR03D5W/p1615381698260700?thread_ts=1614918257.159100&channel=C01MFR03D5W&message_ts=1615381698.260700

**Comment by sherifnada** [comment]
The config from the user 

```
name: "cdp-test-s3"
  sourceDefinitionId: "778daa7c-feaf-4db6-96f3-70fd645acc77"
  workspaceId: "5ae6b09b-fdec-41af-aaf7-7d94cfc33ef6"
  sourceId: "d5982958-0125-46ef-bfff-8042ee48aaa3"
  configuration:
    reader_options: "{ \"sep\" : \",\", \"header\" : 0, \"names\": \"id,first_name,last_name,email,gender,ip_address,dob\"\
      }"
    dataset_name: "employees"
    provider:
      aws_secret_access_key: "*******************************"
      aws_access_key_id: "***********************"
      storage: "S3"
    format: "csv"
    url: "s3://swayam-nc-connector-test-v1/aws_s3_sample.csv"
```

**Comment by yevhenii-ldv** [comment]
The `reader_options` parameter is set incorrectly, the `names` must be a **list of strings**, but in this case it is a **string** and Pandas, when reading, thinks that each character of this string is the name of the column.

The correct configuration of `reader_options` in this case will be as follows:
```
  reader_options: "{ \"sep\" : \",\", \"header\" : 0, \"names\":[\"id\",\"first_name\",\"last_name\",\"email\",\"gender\",\"ip_address\",\"dob\"]}"

```

**Comment by sherifnada** [comment]
@yevhenii-ldv 's suggested solution fixed this problem. This problem should be fixed by implementing #2119 in the long-term, but we can also create an issue to verify that `check` verifies this, maybe by reading 100 rows from the file or something.
